src/servers/grpc_server.cc

```diff
@@ -711,6 +711,7 @@ InferAllocatorPayload(
     AllocPayload* alloc_payload)
 {
   alloc_payload->response_ = &response;
+  alloc_payload->shm_map_ = new AllocPayload::TensorShmMap;
```

Comment:
We don't want to allocate this everytime, and also not for all the requests that don't even use shared memory. So use:
if (alloc_payload->shm_map_ != nullptr) { alloc_payload->shm_map_.clear(); }

Reply:
Fixed